### DISCUSSION TITLE
How can I get rid of snippets

### DISCUSSION THREAD

**Discussion by dist1ll** [hrsh7th/nvim-cmp#759] [General]
The snippets are cluttering my autocomplete. I understand that nvim-cmp only works with a snippet engine. 

For me however, that's a dealbreaker. Could someone point me in the direction of what I need to do to remove them? I've already Ctrl+F removed all references to snippets in the the source code. 

How do I avoid them being listed in the popup? Any help is appreciated.

![image](https://user-images.githubusercontent.com/39585474/151059736-e06dea5b-8750-41c1-9c72-6a0eec5ff3dd.png)

**Comment by dist1ll** [comment]
Figured it out. Replace this line

https://github.com/hrsh7th/nvim-cmp/blob/d93104244c3834fbd8f3dd01da9729920e0b5fe7/lua/cmp/source.lua#L112

with this

```lua
if e:get_kind() ~= 15 then 
  table.insert(entries, e)
end
```

**Comment by piotrostr** [comment]
In case you are using something like this

```lua
-- file: lspconfig.lua
local capabilities = require('cmp_nvim_lsp').update_capabilities(
  vim.lsp.protocol.make_client_capabilities()
)
```

you can disable snippets like this

```lua
capabilities.textDocument.completion.completionItem.snippetSupport = false
```

**Comment by nik-rev** [comment]
The above approach didn't work for me, however I found an alternative.

```lua
{
        name = "nvim_lsp",
},

```

Replaced to:

```lua
{
	name = "nvim_lsp",
	entry_filter = function(entry, ctx)
		return require("cmp").lsp.CompletionItemKind.Snippet ~= entry:get_kind()
	end,
},
```

The issue specifically occured in jsx and tsx files. It would turn anything you type and suggest it as a snippet. Eg:

![image](https://github.com/hrsh7th/nvim-cmp/assets/154856872/43f9b5a4-7a66-4dbb-a252-1bd0f3a0dc71)

**Comment by tkkcc** [comment]
check this https://github.com/zjp-CN/nvim-cmp-lsp-rs or use helix

**Comment by tf42src** [comment]
Same request here. Is there a non-hacky way which is usable via lazyvim's default setup/opts? This is my current config:

```
return {
  {
    "hrsh7th/nvim-cmp",
    opts = {
      experimental = {
        ghost_text = false,
      },
    },
  },
}
```
